envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -268,6 +272,47 @@ private fun Value.toOutgoingTimeoutPolicy(default: Outgoing.TimeoutPolicy): Outg
     return Outgoing.TimeoutPolicy(idleTimeout ?: default.idleTimeout, requestTimeout ?: default.requestTimeout)
 }
 
+private fun Value.toOAuth(properties: SnapshotProperties): OAuth {
+    val provider = this.field("provider").toOauthProvider(properties)
+    val policy = this.field("policy").toOAuthPolicy()
+    val verification = this.field("verification").toOAuthVerification()
+
+    return OAuth(provider, verification, policy)
+}
+
+fun Value?.toOauthProvider(properties: SnapshotProperties) = this?.stringValue
+    ?.takeIf { it.isNotEmpty() }
+    ?.let { provider ->
+        properties.jwt.providers
+            .map { property -> property.name }
+            .firstOrNull {
```

Comment:
Maybe we can build some map so we won't have to iterate each time and just do it in O(1) ?

Reply:
here I'm not really sure how to do it as this is inside an extension function

Reply:
Let's make `providers` a map `<String, ..>` and we won't have to iterate